hbase-server/src/main/java/org/apache/hadoop/hbase/wal/BoundedRecoveredHFilesOutputSink.java

```diff
@@ -191,50 +186,22 @@ public boolean keepRegionEvent(Entry entry) {
     return false;
   }
 
+  /**
+   * @return Returns a base HFile without compressions or encodings; good enough for recovery
```

Comment:
Should we try to get the TableDescriptor first? If it is not possible, then we fallback to write generic HFiles.

Reply:
In Jira, I added a comment about making sure we will compact all these tiny HFiles created as part of WAL split. If we can make sure that part, I would say it ok to create these tiny files with out any table specific things like compression/DBE etc. Anyways we know all these files are going to get compacted and rewritten once we open the region.  
As of now we are not sure whether or when these tiny files will get compacted. In such case I would +1 ur ask. Do this HFile create with defaults as a fall back only.

Reply:
Asking the Master for the table descriptor first and if it fails, read the fs is what we had before this patch. Problem was that default configuration had it that the RPC query wasn't timing out before the below happened:

 2020-06-18 19:53:54,175 ERROR [main] master.HMasterCommandLine: Master exiting
 java.lang.RuntimeException: Master not initialized after 200000ms

I could dick around w/ timings inside here so RPC timed out sooner. It uses the server's Connection. I suppose I could make a new Connection everytime we want to query a Table schema and configure it to try once only but that is a lot of work to do at recovery time.

On the compactions, don't we usually pick up the small files first?

Reply:
Actually the timeout can be set per rpc request on the same connection.

Reply:
bq. Actually the timeout can be set per rpc request on the same connection.

You are right. Could try this later.